src/main/java/com/karankumar/bookproject/ui/shelf/BooksInShelfView.java

```diff
@@ -143,12 +144,18 @@ private HorizontalLayout initializeLayout() {
         Button addShelf = new Button("Add shelf");
         addShelf.addClickListener(e -> customShelfForm.addShelf());
 
+        editShelf = new Button("Edit shelf");
+        editShelf.addClickListener(e -> customShelfForm.editShelf(chosenShelf));
+        // Disable shelf edit by default - should only be enabled when a custom shelf is selected.
+        editShelf.setEnabled(false);
```

Comment:
May be we can add a test for this as well?

Reply:
What should be tested here, simply that clicking on Edit will open the dialog and that the Edit field is disabled by default?

Reply:
@MikeLowrie Both sound good to me

Reply:
Hey sorry for missing this comment, yes agree with @knjk04 's comment here